lib/collaborative_filtering.py

```diff
@@ -105,24 +105,34 @@ def train(self, item_vecs=None):
         Train model for n_iter iterations from scratch.
 
         """
+        matrices_found = False
         if self.load_matrices is False:
             self.user_vecs = numpy.random.random((self.n_users, self.n_factors))
             if item_vecs is None:
                 self.item_vecs = numpy.random.random((self.n_items, self.n_factors))
             else:
                 self.item_vecs = item_vecs
         else:
-            _, self.user_vecs = self.initializer.load_matrix(self.config,
-                                                             'user_vecs', (self.n_users, self.n_factors))
+            users_found, self.user_vecs = self.initializer.load_matrix(self.config,
+                                                                       'user_vecs', (self.n_users, self.n_factors))
             if item_vecs is None:
-                _, self.item_vecs = self.initializer.load_matrix(self.config,
-                                                                 'item_vecs', (self.n_items, self.n_factors))    
+                items_found, self.item_vecs = self.initializer.load_matrix(self.config, 'item_vecs',
+                                                                           (self.n_items, self.n_factors))
             else:
-                self.item_vecs = numpy.random.random((self.n_items, self.n_factors))
-        self.partial_train()
-        if self.dump:
-            self.initializer.save_matrix(self.user_vecs, 'user_vecs')
-            self.initializer.save_matrix(self.item_vecs, 'item_vecs')
+                items_found = True
+                self.item_vecs = item_vecs
+            matrices_found = users_found and items_found
+            if self._v and matrices_found:
+                print("User and Document distributions files found, will not train collaborative.")
```

Comment:
Now for evaluation purposes, I assume that the grid search can retrain n more iterations even if the matrices are loaded so I don't want to switch things around.